Integration tests failures

The integration tests have been failing inconsistently.

```
ERROR: saving image: failed to write image to the following tags: [ttl.sh/test-main:1h: HEAD https://ttl.sh/v2/test-main/manifests/1h: unexpected status code 500 Internal Server Error (HEAD responses have no body, use GET for details)]
Error: Process completed with exit code 1.
```

This usually happens when I'm merging PRs to bump buildpacks for a release (which is the worst time).

It is not clear why it's happening, but it could be something with rate limiting or possibly concurrent access. A number of jobs kick off to run the integration tests because we are merging a number of PRs in succession to prepare for the release.

I can think of a couple options here:

1. We should first try changing the tag that gets generated for ttl.sh. The `test-main` part is the same for all of the jobs, try adding some bit of randomness there. That way if the integration tests are running concurrently, the image being written to is not the same for all of them.
2. Move away from ttl.sh. Use a built-in registry. Run the standard docker registry container on the test VM. Push to and use that.
3. Move away from ttl.sh. Use ghcr.io as temporary storage for the images. Push them with some unique tag for each CI run (so they don't overlap). Then run the tests. Then delete any images from ghcr.io.

Hopefully 1.) fixes it, cause that's the most convenient, but if it does not I'd be leaning towards 3.) as these images are not secret and we should be able to clean them up easily if we tag them in a reasonable way.

--- a/.github/workflows/pb-create-package.yml
+++ b/.github/workflows/pb-create-package.yml
@@ -171,8 +171,11 @@ jobs:
                     TAG="${PACKAGE}-$(mktemp -u XXXXX | awk '{print tolower($0)}'):${VERSION}"
                     pack -v buildpack package "${TAG}" ${CONFIG} --format "${FORMAT}" --publish
                   else
-                    pack -v buildpack package "${PACKAGE}:${VERSION}" ${CONFIG} --format "${FORMAT}"
+                    TAG="${PACKAGE}:${VERSION}"
+                    pack -v buildpack package "${TAG}" ${CONFIG} --format "${FORMAT}"
                   fi
+
+                  echo "ttl-image-tag=${TAG:-}" >> "$GITHUB_OUTPUT"
                 fi
               env:
                 PACKAGES: docker.io/paketobuildpacks/java

--- a/.github/workflows/pb-create-package.yml
+++ b/.github/workflows/pb-create-package.yml
@@ -168,7 +168,7 @@ jobs:
                     done
                 else
                   if [ -n "$TTL_SH_PUBLISH" ] && [ "$TTL_SH_PUBLISH" = "true" ]; then
-                    TAG="${PACKAGE}-$(mktemp -u XXXXX):${VERSION}"
+                    TAG="${PACKAGE}-$(mktemp -u XXXXX | awk '{print tolower($0)}'):${VERSION}"
                     pack -v buildpack package "${TAG}" ${CONFIG} --format "${FORMAT}" --publish
                   else
                     pack -v buildpack package "${PACKAGE}:${VERSION}" ${CONFIG} --format "${FORMAT}"

--- a/.github/workflows/pb-create-package.yml
+++ b/.github/workflows/pb-create-package.yml
@@ -166,11 +166,13 @@ jobs:
                         crane tag "${P}:${VERSION}" latest
                       fi
                     done
-
                 else
-                  pack -v buildpack package \
-                    "${PACKAGE}:${VERSION}" ${CONFIG} \
-                    --format "${FORMAT}" $([ -n "$TTL_SH_PUBLISH" ] && [ "$TTL_SH_PUBLISH" = "true" ] && echo "--publish")
+                  if [ -n "$TTL_SH_PUBLISH" ] && [ "$TTL_SH_PUBLISH" = "true" ]; then
+                    TAG="${PACKAGE}-$(mktemp -u XXXXX):${VERSION}"
+                    pack -v buildpack package "${TAG}" ${CONFIG} --format "${FORMAT}" --publish
+                  else
+                    pack -v buildpack package "${PACKAGE}:${VERSION}" ${CONFIG} --format "${FORMAT}"
+                  fi
                 fi
               env:
                 PACKAGES: docker.io/paketobuildpacks/java

--- a/.github/workflows/pb-tests.yml
+++ b/.github/workflows/pb-tests.yml
@@ -115,6 +115,7 @@ jobs:
                 OS: linux
                 VERSION: ${{ steps.version.outputs.version }}
             - name: Package Buildpack
+              id: package-buildpack
               run: |-
                 #!/usr/bin/env bash
 
@@ -163,8 +164,11 @@ jobs:
                     TAG="${PACKAGE}-$(mktemp -u XXXXX | awk '{print tolower($0)}'):${VERSION}"
                     pack -v buildpack package "${TAG}" ${CONFIG} --format "${FORMAT}" --publish
                   else
-                    pack -v buildpack package "${PACKAGE}:${VERSION}" ${CONFIG} --format "${FORMAT}"
+                    TAG="${PACKAGE}:${VERSION}"
+                    pack -v buildpack package "${TAG}" ${CONFIG} --format "${FORMAT}"
                   fi
+
+                  echo "ttl-image-tag=${TAG:-}" >> "$GITHUB_OUTPUT"
                 fi
               env:
                 FORMAT: image
@@ -182,7 +186,8 @@ jobs:
 
                 set -euo pipefail
 
-                BP_UNDER_TEST=ttl.sh/${PACKAGE}-${VERSION}:1h make integration
+                make integration
               env:
+                BP_UNDER_TEST: ${{ steps.package-buildpack.outputs.ttl-image-tag}}
                 PACKAGE: test
                 VERSION: ${{ steps.version.outputs.version }}

--- a/.github/workflows/pb-tests.yml
+++ b/.github/workflows/pb-tests.yml
@@ -158,15 +158,17 @@ jobs:
                         crane tag "${P}:${VERSION}" latest
                       fi
                     done
-
                 else
-                  pack -v buildpack package \
-                    "${PACKAGE}:${VERSION}" ${CONFIG} \
-                    --format "${FORMAT}" $([ -n "$TTL_SH_PUBLISH" ] && [ "$TTL_SH_PUBLISH" = "true" ] && echo "--publish")
+                  if [ -n "$TTL_SH_PUBLISH" ] && [ "$TTL_SH_PUBLISH" = "true" ]; then
+                    TAG="${PACKAGE}-$(mktemp -u XXXXX):${VERSION}"
+                    pack -v buildpack package "${TAG}" ${CONFIG} --format "${FORMAT}" --publish
+                  else
+                    pack -v buildpack package "${PACKAGE}:${VERSION}" ${CONFIG} --format "${FORMAT}"
+                  fi
                 fi
               env:
                 FORMAT: image
-                PACKAGES: ttl.sh/test-${{ steps.version.outputs.version }}
+                PACKAGES: ttl.sh/paketo-java-test-${{ steps.version.outputs.version }}
                 TTL_SH_PUBLISH: "true"
                 VERSION: 1h
             - name: Set up JDK

--- a/.github/workflows/pb-tests.yml
+++ b/.github/workflows/pb-tests.yml
@@ -160,7 +160,7 @@ jobs:
                     done
                 else
                   if [ -n "$TTL_SH_PUBLISH" ] && [ "$TTL_SH_PUBLISH" = "true" ]; then
-                    TAG="${PACKAGE}-$(mktemp -u XXXXX):${VERSION}"
+                    TAG="${PACKAGE}-$(mktemp -u XXXXX | awk '{print tolower($0)}'):${VERSION}"
                     pack -v buildpack package "${TAG}" ${CONFIG} --format "${FORMAT}" --publish
                   else
                     pack -v buildpack package "${PACKAGE}:${VERSION}" ${CONFIG} --format "${FORMAT}"

